sdk/search/Azure.Search.Documents/src/Indexes/SearchIndexClient.cs

```diff
@@ -0,0 +1,1356 @@
+        /// </param>
+        /// <param name="options">Optional <see cref="SearchRequestOptions"/> to customize the operation's behavior.</param>
+        /// <param name="cancellationToken">Optional <see cref="CancellationToken"/> to propagate notifications that the operation should be canceled.</param>
+        /// <returns>
+        /// The <see cref="Response{T}"/> from the server containing the <see cref="SearchIndex"/> that was created or updated.
+        /// This may differ slightly from what was passed in since the service may return back fields set to their default values depending on the field type and other properties.
+        /// </returns>
+        /// <exception cref="ArgumentNullException">Thrown when <paramref name="index"/> is null.</exception>
+        /// <exception cref="RequestFailedException">Thrown when a failure is returned by the Search service.</exception>
+        public virtual async Task<Response<SearchIndex>> CreateOrUpdateIndexAsync(
+            SearchIndex index,
+            bool allowIndexDowntime = false,
+            bool onlyIfUnchanged = false,
+            SearchRequestOptions options = null,
+            CancellationToken cancellationToken = default)
+        {
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(SearchIndexClient)}.{nameof(CreateOrUpdateIndex)}");
+            scope.Start();
+            try
+            {
+                return await IndexesClient.CreateOrUpdateAsync(
+                    index?.Name,
+                    index,
+                    allowIndexDowntime,
+                    options?.ClientRequestId,
+                    onlyIfUnchanged ? index?.ETag?.ToString() : null,
+                    null,
+                    cancellationToken)
+                    .ConfigureAwait(false);
+            }
+            catch (Exception ex)
+            {
+                scope.Failed(ex);
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Deletes a search index and all the documents it contains.
+        /// </summary>
+        /// <param name="indexName">Required. The name of the <see cref="SearchIndex"/> to delete.</param>
+        /// <param name="options">Optional <see cref="SearchRequestOptions"/> to customize the operation's behavior.</param>
+        /// <param name="cancellationToken">Optional <see cref="CancellationToken"/> to propagate notifications that the operation should be canceled.</param>
+        /// <returns>The <see cref="Response"/> from the server.</returns>
+        /// <exception cref="ArgumentNullException">Thrown when <paramref name="indexName"/> is null.</exception>
+        /// <exception cref="RequestFailedException">Thrown when a failure is returned by the Search service.</exception>
+        public virtual Response DeleteIndex(
+            string indexName,
+            SearchRequestOptions options = null,
+            CancellationToken cancellationToken = default)
+        {
```

Comment:
Nit - would be it be easier to pass `false` for `onlyIfUnchanged` into the overload?  Don't block getting this PR in on it though.

Reply:
I'd have to have a new internal overload that took a string name and `onlyIfUnchanged`, but the scope name would still be the same and it's less to manage should we make changes later.